backend/tracim_backend/lib/utils/request.py

```diff
@@ -61,8 +62,24 @@ def __init__(self) -> None:
         self._client_token = None  # type: typing.Optional[str]
         # Pending events: have been created but are commited to the DB
         self._pending_events = []  # type: typing.List[Event]
+        self._disable_events = False
         self.force_anonymous_context = False
 
+    @property
+    def disable_events(self):
+        return self._disable_events
+
+    @contextlib.contextmanager
+    def batched_events(self, operation_type, obj: object):
+        self._disable_events = True
+        try:
+            yield
+        finally:
+            self._disable_events = False
+            self.plugin_manager.hook.on_batched_events_finished(
+                operation_type=operation_type, obj=obj, context=self
+            )
```

Comment:
The hook should probably executed only if an exception didn't occur:
```suggestion
        try:
            yield
        finally:
            self._disable_events = False
        else:
            self.plugin_manager.hook.on_batched_events_finished(
                operation_type=operation_type, obj=obj, context=self
            )
```